File: frontend/src/types/file.ts
export interface FileEntity {
  id: number
  userId: number
  parentId: number | null
  fileName: string
  fileType: 'file' | 'folder'
  mimeType: string | null
  fileSize: number
  storagePath: string | null
  md5Hash: string | null
  isDeleted: boolean
  versionCount: number
  createdAt: string
  updatedAt: string
}

export interface FileVersionEntity {
  id: number
  fileId: number
  versionNumber: number
  storagePath: string
  fileSize: number
  md5Hash: string | null
  createdAt: string
}

export interface UploadSession {
  sessionId: string
  chunkSize?: number
  totalChunks: number
  uploadedChunks?: number
  missingChunks?: number[]
}

export interface BreadcrumbItem {
  id: number | null
  name: string
}

File: desktop-client/src/betterclouddrive_desktop/models/file.py
from __future__ import annotations

from typing import Literal

from .api import AppModel

FileType = Literal["file", "folder"]


class FileEntity(AppModel):
    id: int
    userId: int
    parentId: int | None = None
    fileName: str
    fileType: FileType
    mimeType: str | None = None
    fileSize: int = 0
    storagePath: str | None = None
    md5Hash: str | None = None
    isDeleted: bool = False
    versionCount: int = 1
    createdAt: str = ""
    updatedAt: str = ""

    @property
    def is_folder(self) -> bool:
        return self.fileType == "folder"


class FileVersionEntity(AppModel):
    id: int
    fileId: int
    versionNumber: int
    storagePath: str
    fileSize: int = 0
    md5Hash: str | None = None
    createdAt: str = ""


class UploadSession(AppModel):
    sessionId: str
    chunkSize: int | None = None
    totalChunks: int = 1
    uploadedChunks: int | None = None
    missingChunks: list[int] | None = None

